src/implementations/twig/components/fact-figures/fact-figures.story.js

```diff
@@ -94,26 +93,32 @@ const getArgTypes = () => ({
 });
 
 const prepareData = (data, args) => {
-  data.items[0].value = args.value;
-  data.items[0].title = args.title;
-  data.items[0].description = args.description;
-  if (args.icon) {
-    data.items[0].icon.name = args.icon;
+  correctPaths(data);
+  const clone = JSON.parse(JSON.stringify(data));
+
+  // Optional elements
+  if (!args.show_view_all) {
+    delete clone.view_all;
   }
-  data.view_all = args.viewAll ? viewAll : {};
-  data.display_icons = args.displayIcons;
-  data.column = args.column;
+  clone.display_icons = args.show_icons;
 
-  if (system === 'ec' && args.displayIcons) {
-    data.items.forEach((item, i) => {
-      data.items[i].icon.size = 'm';
-    });
-    if (args.viewAll) {
-      data.view_all.icon.size = 'xs';
-    }
+  // Column display
+  clone.column = args.column;
+  if (args.column < 3) {
```

Comment:
maybe a ternary operator, here..? you can leave it like it is if you prefer it, eh, there is nothing wrong with it.